spring-integration-core/src/test/java/org/springframework/integration/router/RecipientListRouterTests.java

```diff
@@ -120,20 +121,22 @@ public void sendFailureOnFirstRecipientTriggersExceptionByDefault() {
 		router.setChannels(channels);
 		channelA.send(new GenericMessage<String>("blocker"));
 		Message<String> message = new GenericMessage<String>("test");
-		try {
-			router.handleMessage(message);
-		}
-		catch (RuntimeException e) {
-			Message<?> result1a = channelA.receive(0);
-			assertThat(result1a).isNotNull();
-			assertThat(result1a.getPayload()).isEqualTo("blocker");
-			assertThat(channelB.receive(0)).isNull();
-			assertThat(channelC.receive(0)).isNull();
-			throw e;
-		}
+		assertThatThrownBy(() -> {
+			try {
+				router.handleMessage(message);
+			}
+			catch (RuntimeException e) {
+				Message<?> result1a = channelA.receive(0);
+				assertThat(result1a).isNotNull();
+				assertThat(result1a.getPayload()).isEqualTo("blocker");
+				assertThat(channelB.receive(0)).isNull();
+				assertThat(channelC.receive(0)).isNull();
```

Comment:
Doesn't look like this logic has to be done inside `catch` block.
We simply can move all of these assertions after `assertThatExceptionOfTyle()`.
This way the code in this test would be much easier to read.